services/docs/lib/services/document.js

```
@@ -226,6 +226,41 @@ module.exports.setDocument = function(resource, filter, data, state, user) {
   });
 };
 
+module.exports.patchDocument = (resource, filter, data, state, user) => {
+  return new Promise((resolve, reject) => {
+    builder
+      .patch({
+        resource: resource,
+        data: data,
+        state: state,
+        user: user
+      })
+      .then(update => {
+        console.log(update);
+        resource.collection.findOneAndUpdate(
+          filter,
+          update,
+          { returnDocument: 'after' },
+          (err, result) => {
+            if (err) return reject(err);
+
+            if (!result.value) {
+              return reject(new Error('Not Found'));
+            }
+            resolve({
+              id: filter._id,
+              state: state,
+              data: result.value.states[state].data
+            });
+          }
+        );
+      })
+      .catch(() => {
+        return reject(new Error('Validation Error'));
```

It sounds like this error could directly be thrown by the `builder.patch` function give the fact we're already swallowing the original error to throw this one.